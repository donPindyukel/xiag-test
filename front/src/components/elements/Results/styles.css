.results-wrap {
  h1 {
    font-size: 25px;
    font-weight: normal;
    margin: 0;
    padding: 0;
    line-height: 1.5;
    color: var(--fontColor);
    margin-top: 20px;
  }

  .rows {
    display: flex;
    flex-direction: column;
    width: 400px;
    margin-top: 20px;

    .header,
    .data-row {
      display: flex;
      justify-content: space-between;
    }

    .header {
      text-align: left;
      padding: 10px 20px;
      border-bottom: 1px var(--borderColor) solid;
      background-color: #eee;
      font-size: 16px;
      color: var(--fontColor);
      font-weight: 700;
    }

    .data-row {
      padding: 8px 20px;
      border-bottom: 1px #eee solid;
      font-size: 14px;
      color: var(--fontColor);
    }
  }
}
