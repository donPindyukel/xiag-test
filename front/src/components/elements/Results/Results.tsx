import React from "react";
import { IAnswer, IStatData } from "Models";
import "./styles.css";

export interface IResultsProps {
  answers: IAnswer[];
  statData: IStatData[];
}

export class Results extends React.Component<IResultsProps, {}> {

  render() {
    const data = this.props.statData;
    const answers = this.props.answers;
    return (
      <div className="results-wrap">
        <h1>Results</h1>
        <div className="rows">
          <div className="header">
            <div className="name">Name</div>
            {answers.map((answer: IAnswer) => (
              <div className="answer" key={answer.answerId}>
                {answer.title}
              </div>
            ))}
          </div>
          {data.map((item: IStatData) => (
            <div className="data-row" key={item.userId}>
              <div className="name">{item.userName}</div>
              {answers.map((answer: IAnswer) => (
                <div
                  className="answer-data"
                  key={`${answer.answerId}-data-row`}
                >
                  {answer.answerId === item.answerId ? "x" : ""}
                </div>
              ))}
            </div>
          ))}
        </div>
      </div>
    );
  }
}
