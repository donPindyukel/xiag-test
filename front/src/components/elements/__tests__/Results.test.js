import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Results } from "../Results/Results";

describe("<Results />", () => {
  let component;
  const props = {
    answers: [
      {
        title: 'test',
        answerId: 'test',
      }
    ],
    statData: [{
      userId: 'test',
      userName: 'test',
      answerId: 'test',
      answerTitle: 'test'
    }]
  };

  describe("Results", () => {
    beforeAll(() => {
      component = shallow(<Results {...props}/>);
    });

    it("renders the component", () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});