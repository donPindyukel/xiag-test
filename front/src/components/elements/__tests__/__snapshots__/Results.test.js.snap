// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> Results renders the component 1`] = `
<div
  className="results-wrap"
>
  <h1>
    Results
  </h1>
  <div
    className="rows"
  >
    <div
      className="header"
    >
      <div
        className="name"
      >
        Name
      </div>
      <div
        className="answer"
        key="test"
      >
        test
      </div>
    </div>
    <div
      className="data-row"
      key="test"
    >
      <div
        className="name"
      >
        test
      </div>
      <div
        className="answer-data"
        key="test-data-row"
      >
        x
      </div>
    </div>
  </div>
</div>
`;
